### PR TITLE
Upgrade mocha to latest version (10.0.0)

### PR DESCRIPTION
**Pull request by anmolshres98**
The motivation behind this upgrade is so that we are using a newer version of  `nanoid`. The older version of mocha is using a [vulnerable version of nanoid](https://github.com/advisories/GHSA-qrpm-p2h7-hrv2)

### CODE REVIEW

**Review by ColinKerr — APPROVED** [review]

**Review by mdastous-bentley — APPROVED** [review]

File: test-apps/presentation-test-app/package.json [hunk 1]
```diff
@@ -55,7 +55,8 @@
     "react-dom": "^17.0.0",
     "react-select": "3.2.0",
     "react-resize-detector": "^6.7.6",
-    "semver": "^5.5.0"
+    "semver": "^5.5.0",
+    "mocha": "~10.0.0"
```
Comment (saskliutas) [inline test-apps/presentation-test-app/package.json:59]: Why do test-apps need to have a dependency on `mocha`? I know that there are no tests at least in `presentation-test-app`.

**Review by bsteinbk — COMMENTED** [review]
I don't believe mocha should ever be in the `dependencies` list only in `devDependencies`. I also don't believe it should be added to package files that did not already reference the older version.

**Comment by aruniverse** [comment]
Please also try to keep dependencies sorted. Not sure if you wrote a script to iterate through every package.json and add mocha 10 to it or something similar. Like bill said  no need for it to be added to any new packages here

**Comment by anmolshres98** [comment]
> Please also try to keep dependencies sorted. Not sure if you wrote a script to iterate through every package.json and add mocha 10 to it or something similar. Like bill said no need for it to be added to any new packages here. Also any specific reason for using `~` instead of `^`?

That was my bad on misunderstanding how `rush add` works. I did a `rush add mcoha@latest` which automatically used `~` instead of `^`. Is that a problem @aruniverse ? I will remove the mocha additions to projects that are not needing/using it

**Comment by aruniverse** [comment]
> That was my bad on misunderstanding how `rush add` works. I did a `rush add mcoha@latest` which automatically used `~` instead of `^`. Is that a problem @aruniverse ? I will remove the mocha additions to projects that are not needing/using it

Ah gotcha, i would just do a find all and replace to update deps. This is just a testing tool so i would hope bumping minor versions in a major wouldn't break us, so i think `^` is preferable

**Comment by anmolshres98** [comment]
> > That was my bad on misunderstanding how `rush add` works. I did a `rush add mcoha@latest` which automatically used `~` instead of `^`. Is that a problem @aruniverse ? I will remove the mocha additions to projects that are not needing/using it
> 
> Ah gotcha, i would just do a find all and replace to update deps. This is just a testing tool so i would hope bumping minor versions in a major wouldn't break us, so i think `^` is preferable

ok that sounds good, thanks

**Review by dassaf4 — APPROVED** [review]

**Comment by anmolshres98** [comment]
@saskliutas @aruniverse can I get a review on this after my recent push, thanks

**Review by calebmshafer — APPROVED** [review]

**Review by pmconne — APPROVED** [review]

**Review by bsteinbk — APPROVED** [review]

**Review by saskliutas — APPROVED** [review]

**Comment by anmolshres98** [comment]
@Mergifyio backport release/3.2.x

**Pull request merged**